File: Readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string s = get_string("Text: ");
    double words = 0;
    double letters = 0;
    int len = strlen(s);
    int sentences = 0;
    double average_letters = 0;
    double average_sentences = 0;
    double grade;

    for(int i = 0; i < len; i++){
        if(s[i] == 32 || (s[i] == 33 && s[i + 1] != 32) || (s[i] == 46 && s[i + 1] != 32 && s[i + 1] != 46) || (s[i] == 63 && s[i + 1] != 32) ){
            words++;
        }
     }

    for(int i = 0; i < len; i++){
        if(s[i] != 32 && s[i] != 33 && s[i] != 46 && s[i] != 39 && s[i] != 44 && s[i] != 34 && s[i] != 63){
            letters++;
        }
    }

    for(int i = 0; i < len; i++){
       if(s[i] == 33 || s[i] == 63 || (s[i] == 46 && s[i + 1] != 46)){
            sentences++;
        }
    }

     average_letters =  letters/words * 100;
     average_sentences =  sentences/words * 100;

    grade = round(0.0588 * average_letters - 0.296 * average_sentences - 15.8);

    if(grade >= 16){
        printf("Grade 16+\n");
    }
    else if(grade < 1){
        printf("Before Grade 1\n");
    }
    else{
        printf("Grade %.0f\n", grade);
    }

}

File: Week_6/readability.py
import cs50 as cs

# L = average num of ls per 100 words
# S = average num of sentences per 100 words
# 0.0588 * (L - 0.296) * (S - 15.8)

#This could have been solved without using an array, but i decided to go with this solution
#Reason being: im feeling more comfortable this way, due to my experience with C/C# and arrays in the past.

string = cs.get_string("str: ")
text = [string] #Use string array instead of normal str?
letters = 0
sentences = 0
#Start from 1, because the last word of the sentence wont be counted, because it's not followed by a space.
words = 1

for letter in range(len(text[0])):
    #Function that checks if character is alphabetical
    #Way easier than C
    if text[0][letter].isalpha():
        letters += 1
    #Function that checks if given character is a space, and we count words based on that
    #This is why var words was init from 1 and not 0
    elif text[0][letter].isspace():
        words += 1
    #Checks to see if given char is an exclamation point,question mark or period to count the sentences.
    elif text[0][letter] == chr(46) or text[0][letter] == chr(33) or text[0][letter] == chr(63):
        sentences += 1

average_letters = letters/words * 100
average_sentences = sentences/words * 100

result =  round(0.0588 * average_letters - 0.296 * average_sentences - 15.8)

if result < 1:
    print("Before Grade 1")
elif result > 16:
    print("Grade 16+")
else:
    print(f"Grade {result}")